Mathlib/GameTheory/Auction/Basic.lean

```diff
@@ -58,8 +58,10 @@ logic to handle potential non-constructive cases effectively.
 auction, game theory, economics, bidding, valuation
 -/
 
+open Classical
```

Comment:
This is almost always a bad idea. Can you add the `DecidableEq` on just the lemmas that need it, rather than adding this globally?

Reply:
Hi, Eric. This has been solved, how about the latest version look like.

Reply:
@eric-wieser Hi eric! May I know if there is anything I should modify? 

Next week I am planning to introduce and discuss my plan of formalization of Game Theory in a Workshop at PKU. So it would be helpful to see the first PR is successfully merged. May you please assist me to proceed with it? Grateful!